crud/select/comparators.lua

```diff
@@ -127,24 +128,27 @@ local function array_ge(lhs, rhs, len, lt_funcs, eq_funcs)
     return true
 end
 
+local function cmp_funcs_by_key_type(key_part)
+    local collation = collations.get(key_part)
+    if collations.is_default(collation) then
+        return types.lt(key_part) or lt, types.eq(key_part) or eq
+    elseif collation == collations.UNICODE then
+        return lt_unicode, eq_unicode
+    elseif collation == collations.UNICODE_CI then
+        return lt_unicode_ci, eq_unicode_ci
+    else
+        return nil, GenFuncError:new('Unsupported Tarantool collation %q', collation)
+    end
+end
```

Comment:
I suggest adding `eq_uuid` and `lt_uuid` functions and check that `key_part.type` is `uuid` before checking collation.
It's much more simple and clear solution.

Moreover, I don't like `cmp_funcs_by_key_type` name (in fact, it takes not `key_type` arg but `key_part`). I suggest `get_cmp_func_by_key_part`.

Reply:
I think that collations should be moved to the new types module also.

Reply:
Moved collations to the "types" module